### PR TITLE
Bugfixes/fix gpu validation

### PR DESCRIPTION
**Pull request by baktun14**
- Remove check for mandatory gpu in `v1beta3`
- Added validation from [Cloudmos](https://github.com/akash-network/cloudmos/blob/eca93a683f6e570fe857750f4bf5404e0b4455bf/deploy-web/src/utils/deploymentData/v1beta3.ts#L170-L191)

### CODE REVIEW

File: src/sdl/index.ts [hunk 1]
```diff
@@ -68,42 +69,47 @@ export class SDL {
...
+        throw new Error(`GPU must be one of the supported vendors (${GPU_SUPPORTED_VENDORS.join(",")}).`);
+      }
 
-    if (units > 0 && gpu.attributes?.vendor?.nvidia === undefined) {
-      throw Error("GPU must specify models if units is not 0");
+      const vendor: string = Object.keys(gpu.attributes?.vendor || {})[0];
+
+      if (units > 0 && gpu.attributes?.vendor[vendor] === undefined) {
+        throw new Error("GPU must specify models if units is not 0");
+      }
```
Comment (Redm4x) [inline src/sdl/index.ts:108]: I think the gpu model is optional

File: src/sdl/index.ts [hunk 2]
```diff
@@ -68,42 +69,47 @@ export class SDL {
...
-    const units = parseInt(gpu.units.toString());
+      if (units === 0 && gpu.attributes !== undefined) {
+        throw new Error("GPU must not have attributes if units is 0");
+      }
 
-    if (units == 0 && gpu.attributes !== undefined) {
-      throw new Error("GPU must not have attributes if units is 0");
-    }
+      if (units > 0 && gpu.attributes === undefined) {
+        throw new Error("GPU must not have attributes if units is 0");
```
Comment (Redm4x) [inline src/sdl/index.ts:93]:
```suggestion
        throw new Error("GPU must have attributes if units is not 0");
```
Wrong error message?

Comment (baktun14) [reply src/sdl/index.ts]: Yes

File: src/sdl/index.ts (hunk 1 above)
Comment (baktun14) [reply src/sdl/index.ts]: The test is not necessary now that the test with supported vendors is there

**Pull request merged**